examples/esp-todomvc-react/js/model/eventProcessors/TodoListPostEventProcessor.js

```diff
@@ -0,0 +1,49 @@
+/*global app*/
+
+(function () {
+    "use strict";
+
+    app.model.TodoListPostEventProcessor = {
+        process: function (model) {
+            var filter,
+                id,
+                filteredTodoItems = [],
+                incompleteCount = 0,
+                completeCount = 0,
+                count = 0,
+                allChecked = true,
+                todoItem;
+            if (model.footer.filter === app.model.Filter.complete) {
+                filter = function (todoItem) { return todoItem.complete; };
+            }
+            else if (model.footer.filter === app.model.Filter.active) {
+                filter = function (todoItem) { return !todoItem.complete; };
+            } else {
+                filter = function () { return true; };
+            }
+            for (id in model.todoItemsById) {
+                if (model.todoItemsById.hasOwnProperty(id)) {
+                    todoItem = model.todoItemsById[id];
+                    if (filter(todoItem)) {
+                        filteredTodoItems.push(todoItem);
+                        allChecked = todoItem.complete && allChecked;
+                    }
+
+                    if (!todoItem.complete) {
+                        incompleteCount++;
+                    } else {
+                        completeCount++;
+                    }
+
+                    count++;
+                }
+            }
+            model.main.filteredTodoItems = filteredTodoItems;
+            model.main.toggleAll.visible = filteredTodoItems.length > 0;
```

Comment:
I agree with richard on this one. It's a model for the screen not a pure domain model. 

For example if you wanted to persist the entire state of the screen but some of that state was actually in the components (i.e. isExpanded, isOpen, hasFocus etc) then you'd not be able to get a full picture.

This is a core concept of ESP, it is encouraged to broaden the concept of your domain. More on this here https://github.com/esp/esp-js#single-root-model